fix-frontend:  Rediseño de las pantallas "Agregar tarjetas" y "Mis tarjetas" / Persistencia al eliminar tarjetas en MisTarjetasScreen

EliasRamiroMolinaSerrano opened the pull request:
El diseño se cambio y se aplico el que esta en figma dado a que las anteriores pantallas no seguian este diseño. Ademas de esto se modifico el archivo de agregar tarjetas para que tambien tubiese un estilo similar, sin embargo hay unos problemas que no he podido solucionar cuando intento hacer la pantalla similar y no permite guardar las tarjetas (en esta version se lo permite pero no es tan fiel al diseño). Faltan cosas en el back para que pueda traer los datos y mostrarlos en las tarjetas y que se guarden en la DB. No se entiende la parte donde menciona lo siguiente " Además, no aparece la marca de la tarjeta (Visa, Mastercard, etc.) " dado a que en figma no aparece eso.

EliasRamiroMolinaSerrano committed:
fix-frontend:  Rediseño de las pantallas

El diseño se cambio y se aplico el que esta en figma dado a que las anteriores pantallas no seguian este diseño. Ademas de esto se modifico el archivo de agregar tarjetas para que tambien tubiese un estilo similar, sin embargo hay unos problemas que no he podido solucionar cuando intento hacer la pantalla similar y no permite guardar las tarjetas (en esta version se lo permite pero no es tan fiel al diseño). Faltan cosas en el back para que pueda traer los datos y mostrarlos en las tarjetas y que se guarden en la DB. No se entiende la parte donde menciona lo siguiente " Además, no aparece la marca de la tarjeta (Visa, Mastercard, etc.) " dado a que en figma no aparece eso.

NovaNovon reviewed (changes requested):
Las tarjetas pueden ser agregadas, pero al eliminar una tarjeta, la interfaz de usuario muestra inicialmente el comportamiento correcto (la tarjeta desaparece). Sin embargo, al navegar fuera de la pantalla y regresar, la tarjeta eliminada reaparece.

ARTIC000 committed:
feat: Refactor MisTarjetasScreen to include AsyncStorage for persistent deleted cards and enhance UI components

- Implemented AsyncStorage to persist deleted cards in MisTarjetasScreen.
- Added loading and refreshing states for better user experience.
- Introduced new UI components: Badge, Button, Card, IconButton, Input, Product, and Text for consistent styling.
- Enhanced error handling and loading indicators in fetchTarjetas function.
- Added pull-to-refresh functionality for card list.
- Updated styles and structure for improved maintainability and readability.

ARTIC000 committed:
feat: Update theme management by restructuring theme file and adjusting imports across components

NovaNovon reviewed (approved):
Funciona correctamente

ARTIC000 reviewed (approved):
Funciona bien

killerfrix commented:
This closes #89
